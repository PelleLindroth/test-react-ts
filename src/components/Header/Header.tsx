import { useState } from 'react'
import NavBar from '../NavBar'

const Header = () => {
  const [view, setView] = useState<number>(0)
  const menuItems = ['Home', 'About', 'Contact', 'Profile']

  return (
    <header>
      {`Current view: ${view}`}
      <NavBar items={menuItems} onSetView={setView}/>
    </header>
  )
}

export default Header