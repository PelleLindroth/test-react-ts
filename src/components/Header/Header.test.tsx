import { render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import { mount } from "enzyme"
import Header from "./index"

describe('Header', () => {
  it('mounts Header correctly with NavBar', () => {
    const wrapper = mount(<Header />)

    expect(wrapper).toMatchSnapshot()

    wrapper.unmount()
  })
  it('receives current view from NavBar', () => {
    render(<Header />)

    const currentView = screen.getByText(/current view/i)
    const menuItems = screen.getAllByRole('listitem')

    expect(currentView).toHaveTextContent('0')
    
    userEvent.click(menuItems[2])

    expect(currentView).toHaveTextContent('2')
  })
})