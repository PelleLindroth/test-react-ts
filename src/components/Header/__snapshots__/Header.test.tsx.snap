// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header mounts Header correctly with NavBar 1`] = `
<Header>
  <header>
    Current view: 0
    <NavBar
      items={
        Array [
          "Home",
          "About",
          "Contact",
          "Profile",
        ]
      }
      onSetView={[Function]}
    >
      <nav>
        <ul
          className="menu"
        >
          <li
            className="menu-item selected"
            key="Home"
            onClick={[Function]}
          >
            Home
          </li>
          <li
            className="menu-item "
            key="About"
            onClick={[Function]}
          >
            About
          </li>
          <li
            className="menu-item "
            key="Contact"
            onClick={[Function]}
          >
            Contact
          </li>
          <li
            className="menu-item "
            key="Profile"
            onClick={[Function]}
          >
            Profile
          </li>
        </ul>
      </nav>
    </NavBar>
  </header>
</Header>
`;
